addons/stock/models/stock_quant.py

```diff
@@ -849,17 +861,13 @@ def _get_reserve_quantity(self, product_id, location_id, quantity, product_packa
             could be done and how much the system is able to reserve on it
         """
         self = self.sudo()
-        rounding = product_id.uom_id.rounding
+        rounding = self.env['decimal.precision'].precision_get('Product Unit of Measure')
 
         quants = self._gather(product_id, location_id, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict, qty=quantity)
 
         # avoid quants with negative qty to not lower available_qty
         available_quantity = quants._get_available_quantity(product_id, location_id, lot_id, package_id, owner_id, strict)
 
-        # do full packaging reservation when it's needed
-        if product_packaging_id and product_id.product_tmpl_id.categ_id.packaging_reserve_method == "full":
```

Comment:
Since `packaging_reserve_method` still exists on the product category, I think this is still relevant. Otherwise we should get rid of that field, because that's the only place where it's used.